kernel/ap_start/src/lib.rs

```diff
@@ -54,6 +54,10 @@ pub fn kstart_ap(
         processor_id, cpu_id, _stack_start, _stack_end, nmi_lint, nmi_flags
     );
 
+    // As a cheap precaution
+    #[cfg(target_arch = "aarch64")]
+    irq_safety::disable_interrupts();
```

Comment:
actually this is a great idea to have in general. Can you move it to the top of this function and make it always active for all architectures?